lib/rubocop/cop/style/redundant_begin.rb

```diff
@@ -46,9 +46,11 @@ def autocorrect(node)
 
             corrector.replace(
               range_with_surrounding_space(node.loc.expression),
-              range_with_surrounding_space(
-                child.loc.expression
-              ).source.gsub(/^[ \t]{#{indent_diff}}/, '')
+              range_with_surrounding_space(node.loc.expression)
+                .source
+                .gsub(/\A\s*?begin\s*?(;|$)/, '')
```

Comment:
It's difficult to come up with a simple, elegant solution. I have a possibly acceptable suggestion [here](https://gist.github.com/jonas054/eb08d4563ed56c88a081). Maybe you could start with that and see if you can make it nicer @thirtysixthspan. The `heredoc_ranges` method comes from `lib/rubocop/cop/mixin/autocorrect_alignment.rb` but is slightly altered and should of course not be duplicated. One example fails with my code but I'm not sure if the spec example is correct.